lib/client.ts

```diff
@@ -56,6 +57,7 @@ class HttpClient {
 
     this._config = { ...defaults, ...config };
     this._middleware = [];
+    let httpsAgent = null;
```

Comment:
Can I suggest just a small change? because I'm not a big fan of mutation and 'null' value.
Can this be converted to a ternary?
```suggestion
    const httpsAgent = this._config.agentOptions ? new https.Agent({
        ca: this._config.agentOptions.ca,
        cert: this._config.agentOptions.cert,
        key: this._config.agentOptions.key
      }) : undefined;
```